src/main/java/bussiness_layer/dto/TeachingDTO.java

```diff
@@ -14,15 +15,22 @@
 public class TeachingDTO {
     private Integer id;
     private ProfessorDTO professor;
-    private SortedSet<GroupDTO> groups;
+    private Set<GroupDTO> seminarGroups;
+    private Set<GroupDTO> laboratoryGroups;
     private CourseDTO course;
 
     public TeachingDTO(Teaching entity){
         this.id = entity.getId();
         this.professor = new ProfessorDTO(entity.getProfessor());
-        this.groups = new TreeSet<>();
-        this.groups.addAll(entity.getSeminarGroups().stream().map(GroupDTO::new).collect(Collectors.toList()));
-        this.groups.addAll(entity.getLaboratoryGroups().stream().map(GroupDTO::new).collect(Collectors.toList()));
+        this.seminarGroups = entity.getSeminarGroups().stream().map(GroupDTO::new).collect(Collectors.toSet());
+        this.laboratoryGroups = entity.getLaboratoryGroups().stream().map(GroupDTO::new).collect(Collectors.toSet());
         this.course = new CourseDTO(entity.getCourse());
     }
+
+    public SortedSet<GroupDTO> getAllGroups(){
```

Comment:
Maybe it will be better to create a handler for Teaching and to include this method there, what do you think?